lib/matplotlib/pyplot.py

```diff
@@ -1221,7 +1221,17 @@ def subplot(*args, **kwargs):
                         "and/or 'nrows'.  Did you intend to call subplots()?")
 
     fig = gcf()
-    ax = fig.add_subplot(*args, **kwargs)
+
+    # First, search for an existing subplot with a matching spec.
+    key = SubplotSpec._from_subplot_args(fig, args)
+    subplots = {ax.get_subplotspec(): ax for ax in fig.axes
+                if hasattr(ax, 'get_subplotspec')}
+    ax = subplots.get(key)
```

Comment:
I assume there is only one.
```suggestion
    ax = next(
        (ax for ax in fig.axes
         if hasattr(ax, 'get_subplotspec') and ax.get_subplotspec() == key),
        None)
```